tinygrad/lazy.py

```diff
@@ -31,7 +31,7 @@ def _ast_reduceops(op:LazyOp) -> LazyOp:
   if not src.realized:
     assert isinstance(src.op, LazyOp), "if not src.realized, then src.op must be a LazyOp"
     if MERGE_ELEMENTWISE_INTO_REDUCE and src.optype is BinaryOps and len(src.children) <= 1: src = src.op
-  return LazyOp(op.op, (src,), op.arg)
+  return LazyOp(op.op, (src,), tuple(a.unbind()[0] if isinstance(a, (Variable, MulNode, SumNode)) else a for a in op.arg) if op.arg is not None else None)
```

Comment:
Should this be here? Why ever allow bound variables in ReduceOps?

Reply:
I will go through this PR tomorrow again to make sure everything is good, but I have some clarifying questions about your comments.

so I moved the unbind reduceop.arg logic into the `r()` function, and I think the `arg` of a reduceop can only be `new_shape`. So it cannot be None and cannot be tuples either. I updated the logic to unbind that new_shape before creating LazyOp, and keep the bound st in LazyBuffer.

since reduce arg is new_shape, which is a subset of the original shape, it should suffice to get variables from the shapetracker.

Reply:
Oh sorry, I was thinking movementops. ReduceOp can't be none or tuples. I think the unbind in the _reduce_ops function you have now is correct.